byterun/caml/mlvalues.h

```diff
@@ -25,6 +25,12 @@ typedef intnat value;
 typedef int32 opcode_t;
 typedef opcode_t * code_t;
 
+/* atomic_uintnat is declared as a struct type to prevent accidental
+   accesses that don't use the primitives of platform.h */
+typedef struct atomic_uintnat {
+  volatile uintnat val;
+} atomic_uintnat;
```

Comment:
Putting padding around at least some atomics is a good idea (just created #155). I'd like to see some benchmark showing that it makes a difference, though.

I'm not sure what the point of aligning to 64 bytes is, though. I think you just need to ensure that atomics don't share cache lines accidentally. The default alignment of `uintnat` (either 4 or 8 bytes) ensures that they don't straddle cache line boundaries.